src/main/java/com/rusticisoftware/tincan/RemoteLRS.java

```diff
@@ -223,19 +220,113 @@ public void onHeaders(Response webResp) {
                 }
             }
 
-            if (req.getContentType() != null) {
-                webReq.header("Content-Type", req.getContentType());
+            OutputStreamContentProvider content = new OutputStreamContentProvider();
+            InputStreamResponseListener listener = new InputStreamResponseListener();
+
+            try (OutputStream output = content.getOutputStream()) {
+                if (req.getPartList() == null || req.getPartList().size() <= 0) {
+                    if (req.getContentType() != null) {
+                        webReq.header("Content-Type", req.getContentType());
+                    }
+                    else if (req.getContentType() != "GET") {
+                        webReq.header("Content-Type", "application/octet-stream");
+                    }
+
+                    webReq.content(content).send(listener);
+
+                    if (req.getContent() != null) {
+                        output.write(req.getContent());
+                    }
+
+                    output.close();
+                }
+                else {
+
+                    MultiPartOutputStream multiout = new MultiPartOutputStream(output);
+
+                    webReq.header("Content-Type", "multipart/mixed; boundary=" + multiout.getBoundary());
+                    webReq.content(content).send(listener);
+
+                    if (req.getContentType() != null) {
+                        multiout.startPart(req.getContentType());
+                    } else {
+                        multiout.startPart("application/octet-stream");
+                    }
+
+                    if (req.getContent() != null) {
+                        multiout.write(req.getContent());
+                    }
+
+                    for (HTTPPart part : req.getPartList()) {
+                        multiout.startPart(part.getContentType(), new String[]{
+                                "Content-Transfer-Encoding: binary",
+                                "X-Experience-API-Hash: " + part.getSha2()});
+                        multiout.write(part.getContent());
+                    }
+                    multiout.close();
+                }
             }
-            else {
-                webReq.header("Content-Type", "application/octet-stream");
+
+            Response httpResponse = listener.get(5, TimeUnit.SECONDS);
```

Comment:
This appears to be a pretty fundamental change to how processing was handled before. See the listener removal above, this appears to be setting a "timeout" value and processing after that timeout has happened? What if the server takes longer? What if the server is fast, does it respond immediately? I don't think we can make this fundamental of a change without identifying differences in the approach. Is there a reason the listener methodology was replaced? At the very least if this is unavoidable we need to make the `5` configurable.